### PR TITLE
feat: cosmos proto annotations

### PR DESCRIPTION
**Pull request by aljo242**
Add cosmos proto annotations for:
 - `sdk.Dec`
 - `sdk.Int`
 - addresses
 
 Regenerate proto.
 
 Works with `ignite chain serve` using `feat/sdk-ibc-upgrade` branch

### CODE REVIEW

File: proto/claim/claim_record.proto [hunk 1]
```diff
@@ -4,10 +4,14 @@ package tendermint.spn.claim;
...
+import "cosmos_proto/cosmos.proto";
 
 message ClaimRecord {
-  string address   = 1;
-  string claimable = 2
-      [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int", (gogoproto.nullable) = false];
+  string address   = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+  string claimable = 2 [
+    (gogoproto.nullable)   = false,
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
```
Comment (lubtd) [inline proto/claim/claim_record.proto:13]: Is `(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int"` still required?

**Review by lubtd — COMMENTED** [review]
Thanks for going into this.

 What is the benefit of annotation? Is it a requirement for `0.46`?

File: proto/claim/claim_record.proto (hunk 1 above)
Comment (aljo242) [reply proto/claim/claim_record.proto:13]: Yes the customtype annotations are still required.

I believe these will be used with upcoming changes and optimizations to protobuf comp.  As well as upcoming features like SIGN_MODE_TEXTUAL
- https://github.com/cosmos/cosmos-sdk/blob/0cee47dbd29d9809e48a858742e8b16ad5ea4990/docs/architecture/adr-050-sign-mode-textual.md
- https://github.com/cosmos/cosmos-sdk/blob/0cee47dbd29d9809e48a858742e8b16ad5ea4990/docs/architecture/adr-050-sign-mode-textual-annex1.md

Comment (lubtd) [reply proto/claim/claim_record.proto:13]: Ok I see, let's merge it in this PR 👍

**Review by lubtd — APPROVED** [review]

**Pull request merged**